src/pages/my_page/user_profile/UserOrcid.tsx

```diff
@@ -63,27 +93,32 @@ export const UserOrcid = ({ user }: UserOrcidProps) => {
         });
         if (isSuccessStatus(addOrcidResponse.status)) {
           dispatch(setNotification({ message: t('feedback.success.update_orcid'), variant: 'success' }));
-          cristinPersonQuery.refetch();
+          const orcidCredentials = getOrcidCredentials(history.location.search, orcidInfoResponse.data.id);
+          if (!orcidCredentials) {
+            dispatch(setNotification({ message: t('feedback.error.storing_orcid_credentials'), variant: 'error' }));
+          } else {
+            const postOrcidCredentialsResponse = await postOrcidCredentials(orcidCredentials);
+            if (isErrorStatus(postOrcidCredentialsResponse.status)) {
+              dispatch(setNotification({ message: t('feedback.error.storing_orcid_credentials'), variant: 'error' }));
+            }
+          }
         } else if (isErrorStatus(addOrcidResponse.status)) {
-          dispatch(setNotification({ message: t('feedback.error.update_orcid'), variant: 'success' }));
+          dispatch(setNotification({ message: t('feedback.error.update_orcid'), variant: 'error' }));
         }
       }
-      history.push(UrlPathTemplate.MyPageMyProfile);
       setIsAddingOrcid(false);
+      history.replace(UrlPathTemplate.MyPageMyProfile);
```

Comment:
samme helt øverst i `App.tsx`

```suggestion
      history.replace(UrlPathTemplate.MyPageMyPersonalia);
```